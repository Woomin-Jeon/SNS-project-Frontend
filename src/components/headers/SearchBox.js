import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import func from '../../function';
import logo from '../../files/facebooklogo.png';

const getUserDataFromServer = async (loginState, setLoginState) => {
  const { userStore } = await func.getUsers();

  setLoginState({ ...loginState, user: [...userStore] });
};

function SearchBox({
  loginState,
  setLoginState,
  setSearchState,
}) {
  const [userWriting, setUserWriting] = useState('');
  const history = useHistory();
  const { users } = loginState;

  useEffect(() => {
    getUserDataFromServer(loginState, setLoginState);
  }, []);

  const getUserWriting = (writing) => {
    setUserWriting(writing);
  };

  const searchButtonClicked = () => {
    const searchedNames = users.filter((v) => v.userName === userWriting);

    if (searchedNames.length !== 0) {
      setSearchState({ contents: [...searchedNames], exist: true });
    }
    history.push('/search');
  };

  const searchEnterPressed = async (key) => {
    if (key === 'Enter') {
      await searchButtonClicked();
    }
  }

  const moveToPostPage = () => {
    history.push('/post');
  };

  return (
    <div className="search-box">
      <img className="logo" src={logo} onClick={moveToPostPage} alt="" />
      <input
        className="search-input"
        type="text"
        onChange={(e) => getUserWriting(e.target.value)}
        placeholder="　검색"
        onKeyPress={(e) => searchEnterPressed(e.key)}
      />
      <i className="fas fa-search" onClick={searchButtonClicked} />
    </div>
  );
}

export default SearchBox;
