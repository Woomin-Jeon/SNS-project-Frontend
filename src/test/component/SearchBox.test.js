import React from 'react';
import {shallow} from "enzyme";
import {BrowserRouter} from "react-router-dom";
import SearchBox from "../../components/headers/SearchBox";

describe('SearchBox', () => {
  let loginState;
  let setLoginState;
  let setSearchState;

  it('renders well', () => {
    const component = shallow(
      <BrowserRouter>
        <SearchBox
          loginState={loginState}
          setLoginState={setLoginState}
          setSearchState={setSearchState}
        />
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
