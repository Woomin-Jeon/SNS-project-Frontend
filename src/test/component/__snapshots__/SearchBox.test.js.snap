// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders well 1`] = `ShallowWrapper {}`;
